ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PostExecutionFinalizerWithFingerprintsFromValues.scala

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator.preexecution
+
+import java.time.Instant
+
+import com.daml.ledger.participant.state.kvutils.{Bytes, Err, Fingerprint}
+import com.daml.ledger.participant.state.v1.SubmissionResult
+import com.daml.ledger.validator.LedgerStateOperations
+import com.daml.ledger.validator.LedgerStateOperations.Value
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * An in-transaction post-execution finalizer to be invoked as the last stage of a pre-execution pipeline that
+  * performs deferred time bound checks, detects pre-execution conflicts and persists both state changes and
+  * a suitable log entry.
+  *
+  * It access the ledger state through [[LedgerStateOperations]] that doesn't provide fingerprints along values
+  * and so it is parametric in the logic that produces a fingerprint given a value.
+  *
+  * @param valueToFingerprint The logic producing a [[Fingerprint]] given a value.
+  * @tparam LogResult Type of the offset used for a log entry.
+  */
+class PostExecutionFinalizerWithFingerprintsFromValues[LogResult](
```

Comment:
Can you rename this to something shorter, like `PostExecutionFinalizer`, please? I.e., we most probably won't have two versions of post-execution logic.

Reply:
Done.